### PR TITLE
Integrate `0x99ba64` and `0x99ca64`

### PR DESCRIPTION
**Pull request by marijnvdwerf**
Depends on #4209

### CODE REVIEW

File: src/world/footpath.c [hunk 1]
```diff
@@ -1138,12 +1140,14 @@ static void loc_6A6D7E(
...
-					uint16 di = (mapElement->properties.track.sequence & 0x0F) | (mapElement->properties.track.type << 4);
-					if (!(RCT2_ADDRESS(0x0099CA64, uint8)[di] & (1 << 5))) {
+
+					const uint8 trackType = mapElement->properties.track.type;
+					const uint8 trackSequence = mapElement->properties.track.sequence & 0x0F;
+					if (!(loc_99ca64[trackType][trackSequence] & TRACK_FLAG_5)) {
 						return;
 					}
 					uint16 dx = ((direction - mapElement->type) & 3) ^ 2;
-					if (!(RCT2_ADDRESS(0x0099CA64, uint16)[di / 2] & (1 << dx))) {
```
Comment (marijnvdwerf) [inline src/world/footpath.c]: @duncanspumpkin I'm unsure regarding this change. It makes sense, but I'm not sure how it compares to the original for uneven trackSequences.

File: src/ride/track.h [hunk 2]
```diff
@@ -448,6 +449,24 @@ enum {
...
+	FLAT_TRACK_ELEM_1_X_1_A = 118,
+	FLAT_TRACK_ELEM_1_X_4_B = 119,
+	FLAT_TRACK_ELEM_1_X_1_B = 121,
+	FLAT_TRACK_ELEM_1_X_4_C = 122,
+	FLAT_TRACK_ELEM_3_X_3 = 123,
+};
+
+enum {
+	TRACK_FLAG_4 = (1 << 4), // 0x10
+	TRACK_FLAG_5 = (1 << 5), // 0x20
```
Comment (marijnvdwerf) [inline src/ride/track.h]: Seems to indicate that the element connects directly to paths (only used by `1x1` tiles.

**Comment by IntelOrca** [comment]
This needs rebasing.

**Comment by marijnvdwerf** [comment]
Any suggestions regarding naming? `TRACK_FLAG_4` seems to be used for the first tile in the sequence, `TRACK_FLAG_5` seems to indicate that paths connect to the track part, `TRACK_FLAG_6` apparently has something to do with fences, but I don't really get what.

**Comment by Gymnasiast** [comment]
I think TRACK_FLAG_6 determines if you can build walls on the same tile as that track segment.

**Comment by marijnvdwerf** [comment]
@Gymnasiast Yeah, I figured that one out as well ;)

File: src/world/footpath.c [hunk 3]
```diff
@@ -1138,12 +1140,14 @@ static void loc_6A6D7E(
...
-					if (!(RCT2_ADDRESS(0x0099CA64, uint8)[di] & (1 << 5))) {
+
+					const uint8 trackType = mapElement->properties.track.type;
+					const uint8 trackSequence = mapElement->properties.track.sequence & 0x0F;
+					if (!(FlatRideTrackSequenceProperties[trackType][trackSequence] & TRACK_SEQUENCE_FLAG_CONNECTS_TO_PATH)) {
 						return;
 					}
 					uint16 dx = ((direction - mapElement->type) & 3) ^ 2;
-					if (!(RCT2_ADDRESS(0x0099CA64, uint16)[di / 2] & (1 << dx))) {
+					if (!(FlatRideTrackSequenceProperties[trackType][trackSequence] & (1 << dx))) {
```
Comment (marijnvdwerf) [inline src/world/footpath.c]: I'm unsure regarding this change. It makes sense, but I'm not sure how it compares to the original for uneven trackSequences.

**Pull request merged**